### PR TITLE
Project management - data layer integration

### PR DESCRIPTION
**Pull request by manoramahp**
Resolved the exceptions due to database changes when creating a new project.

### CODE REVIEW

File: modules/components/org.wso2.carbon.ml.database/src/main/java/org/wso2/carbon/ml/database/internal/MLDatabaseService.java [hunk 1]
```diff
@@ -149,7 +149,34 @@ public void insertDatasetDetails(String name, String tenantID, String username,
...
+            result = statement.executeQuery();
+            if (result.first()) {
+                return result.getLong(1);
+            } else {
+                throw new DatabaseHandlerException(
+                        "No dataset id associated with dataset name: " + datasetName + " and tenant id:" + tenantId);
+            }
+        } catch (SQLException e) {
+            throw new DatabaseHandlerException(
+                    " An error has occurred while extracting dataset name: " + datasetName + " and tenant id:" + tenantId);
```
Comment (srinathperera) [inline modules/components/org.wso2.carbon.ml.database/src/main/java/org/wso2/carbon/ml/database/internal/MLDatabaseService.java]: when you rethrow, include the SQL extension as root extension.

File: modules/components/org.wso2.carbon.ml.database/src/main/java/org/wso2/carbon/ml/database/internal/MLDatabaseService.java [hunk 2]
```diff
@@ -149,7 +149,34 @@ public void insertDatasetDetails(String name, String tenantID, String username,
...
+        } finally {
+            // Close the database resources.
+            MLDatabaseUtils.closeDatabaseResources(connection, statement, result);
+        }
+    }
+
+    @Override
```
Comment (srinathperera) [inline modules/components/org.wso2.carbon.ml.database/src/main/java/org/wso2/carbon/ml/database/internal/MLDatabaseService.java]: Method level comments.

File: modules/components/org.wso2.carbon.ml.database/src/main/java/org/wso2/carbon/ml/database/internal/MLDatabaseService.java [hunk 3]
```diff
@@ -246,6 +273,33 @@ public void insertValueSet(String datasetVersionId, String tenantId, String uri,
...
+            statement.setString(2, datasetVersion);
+            result = statement.executeQuery();
+            if (result.first()) {
+                return result.getLong(1);
+            } else {
+                throw new DatabaseHandlerException(
+                        "No dataset id associated with dataset id: " + datasetId + " and version:" + datasetVersion);
+            }
+        } catch (SQLException e) {
+            throw new DatabaseHandlerException(
```
Comment (srinathperera) [inline modules/components/org.wso2.carbon.ml.database/src/main/java/org/wso2/carbon/ml/database/internal/MLDatabaseService.java]: Propergate information about root cause exception up.

File: modules/components/org.wso2.carbon.ml.database/src/main/java/org/wso2/carbon/ml/database/internal/MLDatabaseService.java [hunk 4]
```diff
@@ -614,10 +668,10 @@ public void createProject(String projectID, String projectName, String descripti
             connection = dbh.getDataSource().getConnection();
             connection.setAutoCommit(false);
             createProjectStatement = connection.prepareStatement(SQLQueries.CREATE_PROJECT_NEW);
-            createProjectStatement.setString(1, projectID);
-            createProjectStatement.setString(2, projectName);
-            createProjectStatement.setString(3, description);
-            createProjectStatement.setString(4, String.valueOf(tenantId));
+            //createProjectStatement.setString(1, projectID);
```
Comment (srinathperera) [inline modules/components/org.wso2.carbon.ml.database/src/main/java/org/wso2/carbon/ml/database/internal/MLDatabaseService.java]: cleanup

**Comment by manoramahp** [comment]
Fixes : https://github.com/wso2/product-ml/pull/67

File: modules/components/org.wso2.carbon.ml.database/src/main/java/org/wso2/carbon/ml/database/internal/MLDatabaseService.java (hunk 1 above)
Comment (manoramahp) [inline modules/components/org.wso2.carbon.ml.database/src/main/java/org/wso2/carbon/ml/database/internal/MLDatabaseService.java]: committed in https://github.com/wso2/product-ml/commit/56497b808effb296cc55d6800b2326ce266e6b8a

File: modules/components/org.wso2.carbon.ml.database/src/main/java/org/wso2/carbon/ml/database/internal/MLDatabaseService.java (hunk 2 above)
Comment (manoramahp) [inline modules/components/org.wso2.carbon.ml.database/src/main/java/org/wso2/carbon/ml/database/internal/MLDatabaseService.java]: Added comments : 
https://github.com/wso2/product-ml/commit/9763ede6ebfda2fb8ff061e50f9dca236f1f7b6e
https://github.com/wso2/product-ml/commit/0d63165a9f1eab05b9924d43c656ed5f7a33e2b9

File: modules/components/org.wso2.carbon.ml.database/src/main/java/org/wso2/carbon/ml/database/internal/MLDatabaseService.java (hunk 4 above)
Comment (manoramahp) [inline modules/components/org.wso2.carbon.ml.database/src/main/java/org/wso2/carbon/ml/database/internal/MLDatabaseService.java]: committed : 
https://github.com/wso2/product-ml/commit/d4f36e3ee8c2a10bc526511bd93ebfdae369e6fb
https://github.com/wso2/product-ml/commit/cd527f3daf1098d641d98f96708f83547d3f275e

**Pull request merged**